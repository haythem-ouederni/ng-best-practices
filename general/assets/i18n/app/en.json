{
  "app": {
    "env_message": "Production environment : {{env}}",
    "title": "Welcome to {{ title }}!",
    "title_2": "Angular Best Practices Example",
    "buttons": {
      "go_to_page_one": "Go to page one"
    }
  }
}