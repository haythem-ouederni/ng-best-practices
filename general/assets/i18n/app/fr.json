{
  "app": {
    "env_message": "Environnement de production : {{env}}",
    "title": "Bienvenue à {{ title }}!",
    "title_2": "Angular Best Practices Exemple",
    "buttons": {
      "go_to_page_one": "Aller à la première page"
    }
  }
}