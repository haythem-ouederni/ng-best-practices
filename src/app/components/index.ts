export * from './welcome';
export * from './main';
