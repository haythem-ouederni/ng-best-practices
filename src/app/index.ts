export * from './components';
// export * from './lazy-modules'; => should not be added to the barrel we have an unexpected exception
export * from './app-routing.module';
export * from './app.module';
