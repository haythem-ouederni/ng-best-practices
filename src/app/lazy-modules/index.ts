export * from './lazy-connection.module';
