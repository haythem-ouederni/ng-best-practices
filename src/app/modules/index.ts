export * from './app-routing.module';
export * from './app.module';
