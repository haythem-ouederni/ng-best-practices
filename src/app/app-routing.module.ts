import {NgModule} from '@angular/core';
import {RouterModule, Routes} from '@angular/router';
import {environment} from '../environments/environment';
import {WelcomeComponent} from './components';

const routes: Routes = [
  {
    path: environment.connection.basePath,
    loadChildren: './lazy-modules/lazy-connection.module#LazyConnectionModule',
  },
  {
    path: '',
    component: WelcomeComponent,
  },
];

@NgModule({
  imports: [RouterModule.forRoot(routes)],
  exports: [RouterModule],
})
export class AppRoutingModule {}
