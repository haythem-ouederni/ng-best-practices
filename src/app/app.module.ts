import {BrowserModule} from '@angular/platform-browser';
import {NgModule} from '@angular/core';

import {AppRoutingModule} from './app-routing.module';
import {AppComponent} from './components';
import {WelcomeComponent} from './components/welcome/welcome.component';

@NgModule({
  declarations: [AppComponent, WelcomeComponent],
  imports: [BrowserModule, AppRoutingModule],
  providers: [],
  bootstrap: [AppComponent],
})
export class AppModule {}
