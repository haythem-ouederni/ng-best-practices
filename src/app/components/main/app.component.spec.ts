import {async, ComponentFixture, TestBed} from '@angular/core/testing';
import {RouterTestingModule} from '@angular/router/testing';
import {AppComponent} from './app.component';

describe('AppComponent', () => {
  let component: AppComponent;
  let fixture: ComponentFixture<AppComponent>;

  beforeEach(async(() => {
    TestBed.configureTestingModule({
      imports: [RouterTestingModule],
      declarations: [AppComponent],
    }).compileComponents();
  }));

  beforeEach(() => {
    fixture = TestBed.createComponent(AppComponent);
    component = fixture.debugElement.componentInstance;
    fixture.detectChanges();
  });

  test('should be created', () => {
    expect(component).toBeTruthy();
  });

  test('should match snapshot', () => {
    expect(fixture).toMatchSnapshot();
  });
});
