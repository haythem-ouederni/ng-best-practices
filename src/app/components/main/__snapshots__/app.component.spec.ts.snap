// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppComponent should match snapshot 1`] = `
<app-root>
  <router-outlet />
</app-root>
`;
